### PR TITLE
Use configuration from the editor AND do not expect language server options from the outside anymore

### PR DESCRIPTION
**Pull request by CGNonofr**

### CODE REVIEW

File: src/createLanguageClient.ts [hunk 1]
```diff
@@ -43,12 +43,22 @@ async function openConnection (url: URL | string, errorHandler: ConnectionErrorH
...
+                if (alreadyExisting) {
+                  console.warn('Registration already existing', registration.id, registration.method)
+                }
+                return !alreadyExisting
               })
               registrationParams.registrations.forEach(registration => {
                 existingRegistrations.add(registration.id)
               })
             }
+            if ((args[0] as MessageSignature).method === UnregistrationRequest.type.method) {
```
Comment (MaximeCheramy) [inline src/createLanguageClient.ts:56]: nit: `args[0] as MessageSignature).method` could be extracted into a variable to simplify the 2 conditions.

File: src/createLanguageClient.ts [hunk 2]
```diff
@@ -85,19 +85,19 @@ async function openConnection (url: URL | string, errorHandler: ConnectionErrorH
 const RETRY_DELAY = 3000
 class CGLSPConnectionProvider implements IConnectionProvider {
   private readonly serverUrl: string
-  private readonly language: string
+  private readonly id: string
```
Comment (MaximeCheramy) [inline src/createLanguageClient.ts]: nit: I'm wondering if the type could be an enum of possible values instead of a string. What do you think?

Comment (CGNonofr) [reply src/createLanguageClient.ts]: It's a good idea, I already though about it

File: src/createLanguageClient.ts (hunk 1 above)
Comment (CGNonofr) [reply src/createLanguageClient.ts]: https://github.com/CodinGame/monaco-languageclient-wrapper/pull/4/commits/b2f61ab64de91e1f6e09dc483b61fc7834d04480

File: src/createLanguageClient.ts (hunk 2 above)
Comment (CGNonofr) [reply src/createLanguageClient.ts]: https://github.com/CodinGame/monaco-languageclient-wrapper/pull/4/commits/67551ffb2b457526b0fbd6382388b22c647e75ee

**Review by MaximeCheramy — APPROVED** [review]

**Pull request merged**